pkg/util/converter/prom.go

```diff
@@ -69,14 +107,26 @@ func ReadPrometheusStyleResult(iter *jsoniter.Iterator, opt Options) backend.Dat
 	return rsp
```

Comment:
```suggestion
    if iter.Error != nil && rsp.Error == nil {
			rsp.Error = iter.Error
	}
	return rsp
```
I wonder if the explicit checks are needed within each element, or just checking once at the end 🤦🏻

Reply:
I think it is best to stop as soon as the error as occurred, or is harder to predicate the behavior and later errors may mask previous errors.